ha/cli/command_factory.py

```diff
@@ -18,22 +18,70 @@
 import inspect
 
 from ha.cli import commands
+from ha.core.config.config_manager import ConfigManager
+#from cortx.utils.log import Log
 
-class CommandFactory:
-    """
-    Factory for representing and creating command objects using
-    a generic skeleton.
-    """
 
-    @staticmethod
-    def get_command(component_parser):
+class cmdFactory:
+    def __init__(self):
         """
-        Get command from command factory
-
-        Args:
-            component_parser (<class 'argparse._SubParsersAction'>): argparse to add subparser.
+        init of command factory
         """
-        command_class = inspect.getmembers(commands, inspect.isclass)
-        for cmd_name, cmd in command_class:
-            if cmd_name != "Command" and "Command" in cmd_name:
-                cmd.add_args(component_parser)
+        # Initialize logging.
+        # Prefix of the log file name "cortxcli" is passed to the init function;
+        # So the generated log file will be cortxcli.log
+        ConfigManager.init("cortxcli")
+
+        # This dictionary contains the class names used by all HA CLIs
+        # In case of any modifications to the CLI,
+        # this dictionary should be updated.
+        self.cmd_dict = {
+            "cluster": {
+                "start": "ClusterStartExecutor",
+                "stop": "ClusterStopExecutor",
+                "restart": "ClusterRestartExecutor",
+                "standby": "ClusterStandbyExecutor",
+                "active": "ClusterActiveExecutor",
+                "list": "ClusterListExecutor",
+                "status": "ClusterStatusExecutor",
+                "add": "ClusterAddExecutor",
+                "-h": "CLIUsage",
```

Comment:
Since the flow will come here only if user has executed "cortx cluster -h", the help can be made specific to the cluster it self and displaying complete help will not be required. It applies to all the elements.

Reply:
okay. Will change

Reply:
Changes are done